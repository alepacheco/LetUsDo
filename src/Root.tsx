import React from 'react';
import { History } from 'history';
import PropTypes from 'prop-types';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';
import App from 'src/App';
import initGa from 'src/utils/analytics';

export const Root: React.FC<{ store: any; history: History }> = ({ store, history }) => {
  initGa(history);

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>
  );
};

Root.propTypes = {
  store: PropTypes.object.isRequired
};

export default Root;
