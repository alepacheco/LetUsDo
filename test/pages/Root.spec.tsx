/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Root from '../../src/Root';

jest.mock('react-ga', () => ({
  initialize: jest.fn(),
  set: jest.fn(),
  pageview: jest.fn()
}));

describe('<Root />', () => {
  it('renders', () => {
    const history: any = {
      listen: jest.fn(),
      location: {
        pathname: '/'
      }
    };

    const store = {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn()
    };
    const wrapper = shallow(<Root store={store} history={history} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
