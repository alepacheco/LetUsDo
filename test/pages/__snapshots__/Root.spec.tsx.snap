// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> renders 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {
          "pathname": "/",
        },
      }
    }
  >
    <App />
  </Connect(ConnectedRouterWithContext)>
</Provider>
`;
